src/profile/v100/app_preferences/gateways/saved_gateways.rs

```diff
@@ -66,24 +66,34 @@ impl<'de> Deserialize<'de> for SavedGateways {
         struct Wrapper {
             #[serde(rename = "current")]
             url: Url,
-            saved: IdentifiedVecOf<Gateway>,
+            saved: Vec<Gateway>,
         }
         let wrapped = Wrapper::deserialize(deserializer)?;
         let current = wrapped
             .saved
             .iter()
             .find(|g| g.id() == wrapped.url)
-            .clone()
             .ok_or({
                 CommonError::InvalidGatewaysJSONCurrentNotFoundAmongstSaved
             })
             .map_err(de::Error::custom)?;
 
-        let mut other = wrapped.saved.clone();
+        let saved = wrapped.saved.clone();
+        let mut other = IdentifiedVecOf::<Gateway>::new();
+        for item in saved {
+            let _ = other.try_insert_unique(item.clone()).map_err(|e| {
```

Comment:
The map_err seems superflous, rather I think you wanted to match on the result and log the error for debug purpose